python/ray/_private/worker.py

```diff
@@ -159,9 +163,9 @@ def __init__(self, function: Callable[[T0, T1], R]) -> None:
 
     def remote(
         self,
-        __arg0: "Union[T0, ObjectRef[T0]]",
-        __arg1: "Union[T1, ObjectRef[T1]]",
-    ) -> "ObjectRef[R]":
+        __arg0: "Union[T0, RayWaitable[T0]]",
```

Comment:
I think for the inputs, it should still be T0 or ObjectRef[T0]?

Reply:
Yep updated. I overloaded the remote method so that it can return both generator and object ref. After that it is fixed